src/modules/dashboard/sagas/actions.ts

```diff
@@ -873,6 +873,170 @@ function* createDomainAction({
   return null;
 }
 
+function* editDomainAction({
+  payload: {
+    colonyAddress,
+    colonyName,
+    domainName,
+    domainColor,
+    domainPurpose,
+    annotationMessage,
+    domainId,
+  },
+  meta: { id: metaId, history },
+  meta,
+}: Action<ActionTypes.COLONY_ACTION_DOMAIN_EDIT>) {
+  let txChannel;
+  try {
+    const apolloClient = TEMP_getContext(ContextModule.ApolloClient);
+
+    if (!domainId) {
+      throw new Error('A domain id is required to edit domain');
+    }
+
+    /*
+     * Upload domain metadata to IPFS
+     */
+    let domainMetadataIpfsHash = null;
+    domainMetadataIpfsHash = yield call(
+      ipfsUpload,
+      JSON.stringify({
+        domainName,
+        domainColor,
+        domainPurpose,
+      }),
+    );
+
+    /*
+     * Upload annotationMessage to IPFS
+     */
+    let annotationMessageIpfsHash = null;
+    if (annotationMessage) {
+      annotationMessageIpfsHash = yield call(
+        ipfsUpload,
+        JSON.stringify({
+          annotationMessage,
+        }),
+      );
+    }
```

Comment:
Please have a look at how I refactored these in #2359 

https://github.com/JoinColony/colonyDapp/blob/fixes/qa-initial-stabilization/src/modules/dashboard/sagas/actions.ts#L171-L184

The idea here is basically:
- you move them below, next to the actual transaction in which they are being called
- you put the transaction in a pending state
- then you send the upload action
- then you update the transaction's params with the ipfs hash
- then you send the actual transaction

This makes it so that the transaction will show a loading indicator as soon as it starts to upload the ipfs data.

This is not noticeable on local, but it's a problem on QA, where, the UI will be blocked until IPFS finishes the upload. If you refactor this as I've proposed, you'll already inform the user that something going on in the background.

If you have more questions let me know and I'll come help :)